src/backend/marsha/core/api/playlist.py

```diff
@@ -134,3 +137,26 @@ def destroy(self, request, *args, **kwargs):
             return super().destroy(request, *args, **kwargs)
         except ProtectedError:
             return Response("Resources are still attached to playlist", status=400)
+
+    @action(detail=True, methods=["post"])
+    # pylint: disable=unused-argument
+    def claim(self, request, *args, **kwargs):
+        """Claim a playlist by creating a PlaylistAccess."""
+        playlist = self.get_object()
+        playlist_access_exists = PlaylistAccess.objects.filter(
+            playlist=playlist,
+        ).exists()
+        PlaylistAccess.objects.get_or_create(
+            playlist=playlist,
+            user_id=request.user.id,
+            defaults={
+                "role": INSTRUCTOR if playlist_access_exists else ADMINISTRATOR,
+            },
+        )
+
+        # should we set playlist organization to user organization?
```

Comment:
A playlist can't be both in an organization and in a consumer_site. To do that we must remove the `consumer_site_or_organization_check_idx` index on the playlist model. 
I don't really understand why this constraint exists. It can be useful IMO to have a playlist in both an organization and in a consumer site.

Reply:
Do you think we should do it in this PR, or in a new one ?